default.nix

```diff
@@ -21,7 +21,7 @@ let
     git
     gitAndTools.hub
     nix-prefetch-git
-    nixStable
+    nixUnstable
```

Comment:
Or maybe it should not even manually wrap any `nix` in here since whatever version the user is running would be more correct to run against the `nix-daemon`

Reply:
I agree with that. You should definitely be using the version they have installed, or else you risk miscommunication with the daemon

Reply:
Hm I haven't encountered this directly. For example, `nix-shell -p nixStable2` on NixOS 17.09 (still using nix 1) works fine and even nested shells. However, I'm no expert on how the daemon works so I could see that being a problem.

Reply:
Opening a nix-shell isn't exactly a comprehensive test of all the nix-daemon's capabilities :P I've had breakage in corner cases between 2.0.x bumps, so I'm cautious to use different versions in any scenario. There's nothing stopping them from making major breaking changes to the daemon's protocol in the future, as I don't think it's considered public API.

Reply:
Yeah if there's legitimate concern then I'm all for avoiding issues.